Sources/AWSLambdaRuntimeCore/LambdaHandler.swift

```diff
@@ -44,45 +46,129 @@ public protocol LambdaHandler: EventLoopLambdaHandler {
     ///
     /// - Returns: A Lambda result ot type `Output`.
     func handle(_ event: Event, context: LambdaContext) async throws -> Output
+
+    /// Encode a response of type ``Output`` to `ByteBuffer`.
+    /// Concrete Lambda handlers implement this method to provide coding functionality.
+    /// - parameters:
+    ///     - value: Response of type ``Output``.
+    ///     - buffer: A `ByteBuffer` to encode into, will be overwritten.
+    ///
+    /// - Returns: A `ByteBuffer` with the encoded version of the `value`.
+    func encode(value: Output, into buffer: inout ByteBuffer) throws
+
+    /// Decode a `ByteBuffer` to a request or event of type ``Event``.
+    /// Concrete Lambda handlers implement this method to provide coding functionality.
+    ///
+    /// - parameters:
+    ///     - buffer: The `ByteBuffer` to decode.
+    ///
+    /// - Returns: A request or event of type ``Event``.
+    func decode(buffer: ByteBuffer) throws -> Event
 }
 
 @available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
 extension LambdaHandler {
-    public static func makeHandler(context: LambdaInitializationContext) -> EventLoopFuture<Self> {
-        let promise = context.eventLoop.makePromise(of: Self.self)
+    public init() {
+        self.init()
+    }
+
+    /// The Lambda initialization method.
+    /// Use this method to initialize resources that will be used in every request.
+    ///
+    /// Examples for this can be HTTP or database clients.
+    /// - parameters:
+    ///     - context: Runtime ``LambdaInitializationContext``.
+    public init(context: LambdaInitializationContext) async throws {
+        self.init()
+    }
+}
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+final class CodableLambdaHandler<Underlying: LambdaHandler>: ByteBufferLambdaHandler {
```

Comment:
I would suggest a name change here to avoid confusion with Swift’s `Codable`, but I’ve got no suitable name yet and it’s `internal` so we indeed can correct it later.

Reply:
this is internal, so should not be an issue